Fix silent fail on login

bezumkin opened the pull request:
### What does it do?
Return error if user has no permission for using manager

### Why is it needed?
If user successfully login to manager, but has no permission to use backend, login pagewill  just reload without any errors. That can be very confusing.

### Related issue(s)/PR(s)
#12706

bezumkin committed:
Fix silent fail on login

Return error if user has no permissions for using manager

superspace commented:
I'm testing your patch and got the following result:

1. After a login with a user without permissions to use the backend i get the message "Login failed".
2. I enter the username and password of a user with all required permissions and hit the login button – but the only thing happening is a page reload. No message, no successful login. 
3. I repeat the first step and i get again the message "Login failed".
4. I try another (incorrect) password and hit the login button – the only thing happening is a page reload without a message
#modxbughunt

bezumkin committed:
Prevent reload of login page

If you can't login to manager you will see an error, but the next attempt will just reload window. It is because user is actually logged it but can't see backend, so manager controller will check it and logout him on the next request. That is why we have a page reload instead of new attempt to login.

This change will fix it.

bezumkin commented:
If you can't login to manager you will see an error, but the next attempt will just reload window. It is because user is actually logged it but can't see backend, so manager controller will check it and logout him on the next request. That is why we have a page reload instead of new attempt to login.

I made a new commit, please, check it.

superspace commented:
I checked it again and now it behaves as i would expect it.
